.github/workflows/documentation.yml

```diff
@@ -0,0 +1,27 @@
+name: Generate terraform docs
+on: [push, pull_request]
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
I don't think this will work for push events. See [here](https://github.com/terraform-docs/gh-actions#auto-commit-changes)

Reply:
Can I show you a sample, to check whats missing, on a trial the README.md, the run was automatically updated. since I combined the trigger for both PR and push